### PR TITLE
[Maintenance] Bump Sylius resource packages to ^1.11

### PR DESCRIPTION
**Pull request by NoResponseMate**
| Q               | A
|-----------------|-----
| Branch?         | 1.14
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | -
| License         | MIT

### CODE REVIEW

File: phpstan.neon.dist [hunk 1]
```diff
@@ -74,12 +45,4 @@ parameters:
...
-        - '/PHPDoc tag @extends contains generic type Sylius\\Component\\Resource\\\w+\\\w+<Sylius\\Component\\\w+\\Model\\\w+> but interface Sylius\\Component\\Resource\\\w+\\\w+ is not generic\./'
-        - '/PHPDoc tag @param for parameter \$(\w+) contains generic type Sylius\\Component\\Resource\\\w+\\\w+<Sylius\\Component\\\w+\\Model\\\w+> but interface Sylius\\Component\\Resource\\\w+\\\w+ is not generic\./'
-        - '/PHPDoc type for property Sylius\\Bundle\\\w+\\\w+\\\w+::\$(\w+) contains generic type Sylius\\Component\\Resource\\\w+\\\w+<Sylius\\Component\\\w+\\Model\\\w+> but interface Sylius\\Component\\Resource\\\w+\\\w+ is not generic\./'
+        - '/Method Sylius\\Bundle\\CoreBundle\\Fixture\\OptionsResolver\\LazyOption\:\:(\w+)\(\) has parameter \$repository with generic interface Sylius\\Resource\\Doctrine\\Persistence\\RepositoryInterface but does not specify its types\: T/'
```
Comment (GSadee) [inline phpstan.neon.dist]: Probably there could be defined a template using `ResourceInterface` to resolve this issue

File: src/Sylius/UPGRADE-1.14.md
Comment (GSadee) [inline src/Sylius/UPGRADE-1.14.md:1]: Why are you removing this file?

File: src/Sylius/Component/Product/Factory/ProductFactory.php [hunk 2]
```diff
@@ -14,12 +14,24 @@
...
 {
-    public function __construct(private FactoryInterface $factory, private FactoryInterface $variantFactory)
-    {
+    /**
+     * @param FactoryInterface<T> $factory
+     * @param FactoryInterface<ProductVariantInterface> $variantFactory
+     */
+    public function __construct(
+        private FactoryInterface $factory,
+        private FactoryInterface $variantFactory
```
Comment (GSadee) [inline src/Sylius/Component/Product/Factory/ProductFactory.php:33]:
```suggestion
        private FactoryInterface $variantFactory,
```

**Review by GSadee — DISMISSED** [review]

File: src/Sylius/UPGRADE-1.14.md
Comment (NoResponseMate) [reply src/Sylius/UPGRADE-1.14.md:1]: My guess is it was wrongly added, it's in the wrong place and we have another `UPGRADE-1.14.md` where all the others are placed

File: phpstan.neon.dist (hunk 1 above)
Comment (NoResponseMate) [reply phpstan.neon.dist]: I'll take a look if it can be solved like that 👋

File: src/Sylius/Component/Product/Factory/ProductFactory.php (hunk 2 above)
Comment (GSadee) [reply src/Sylius/Component/Product/Factory/ProductFactory.php:33]: I'm resolving as it will be resolved by the refactor build 🖖🏻

**Review by GSadee — APPROVED** [review]

**Comment by GSadee** [comment]
Thanks, Jan! :1st_place_medal:

**Pull request merged**